posixlib/src/main/scala/scala/scalanative/posix/signal.scala

```diff
@@ -0,0 +1,411 @@
+  type sigaction = CStruct4[
+    CFuncPtr1[CInt, Unit], // sa_handler Ptr to a signal-catching function or one of the SIG_IGN or SIG_DFL
+    sigset_t,              // sa_mask Set of signals to be blocked during execution of the signal handling func
+    CInt,                  // sa_flags Special flags
+    CFuncPtr3[CInt,
+              Ptr[siginfo_t],
+              Ptr[Byte],
+              Unit] // sa_sigaction Pointer to a signal-catching function
+  ]
+
+  // define the following macros which shall expand to integer constant expressions
+  // that need not be usable in #if preprocessing directives
+  @name("scalanative_sig_block")
+  def SIG_BLOCK: CInt = extern
+  @name("scalanative_sig_unblock")
+  def SIG_UNBLOCK: CInt = extern
+  @name("scalanative_sig_setmask")
+  def SIG_SETMASK: CInt = extern
+
+  // define the following symbolic constants
+  @name("scalanative_sa_nocldstop")
+  def SA_NOCLDSTOP: CInt = extern
+  @name("scalanative_sa_onstack")
+  def SA_ONSTACK: CInt = extern
+  @name("scalanative_sa_resethand")
+  def SA_RESETHAND: CInt = extern
+  @name("scalanative_sa_restart")
+  def SA_RESTART: CInt = extern
+  @name("scalanative_sa_siginfo")
+  def SA_SIGINFO: CInt = extern
+  @name("scalanative_sa_nocldwait")
+  def SA_NOCLDWAIT: CInt = extern
+  @name("scalanative_sa_nodefer")
+  def SA_NODEFER: CInt = extern
+  @name("scalanative_ss_onstack")
+  def SS_ONSTACK: CInt = extern
+  @name("scalanative_ss_disable")
+  def SS_DISABLE: CInt = extern
+  @name("scalanative_minsigstksz")
+  def MINSIGSTKSZ: CInt = extern
+  @name("scalanative_sigstksz")
+  def SIGSTKSZ: CInt = extern
+
+  // A machine-specific representation of the saved context
+  // mac OS type mcontext_t = Ptr[__darwin_mcontext64]
+  // __darwin_mcontext64 -> _STRUCT_MCONTEXT64 -> typedef _STRUCT_MCONTEXT64	*mcontext_t;
+  type mcontext_t = Ptr[Byte]
+
+  type ucontext_t = CStruct4[
+    Ptr[Byte], // ucontext_t *uc_link Ptr to the context that is resumed when this context returns (Ptr instead)
+    sigset_t,  // c_sigmask The set of signals that are blocked when this context is active
+    Ptr[
+      stack_t
+    ],         // uc_stack The stack used by this context (Ptr instead of value)
+    mcontext_t // uc_mcontext A machine-specific representation of the saved context
+  ]
+
+  type stack_t = CStruct3[
+    Ptr[Byte], // void *ss_sp Stack base or pointer
+    size_t,    // ss_size Stack size
+    CInt       // ss_flags Flags
+  ]
+
+  type siginfo_t = CStruct9[
+    CInt,       // si_signo Signal number
+    CInt,       // si_code Signal code
+    CInt,       // si_errno If non-zero, an errno value associated with this signal, as described in <errno.h>
+    pid_t,      // si_pid Sending process ID
+    uid_t,      // si_uid Real user ID of sending process
+    Ptr[Byte],  // void *si_addr Address of faulting instruction (Ptr instead - is void * a fcn?)
+    CInt,       // si_status Exit value or signal
+    CLong,      // si_band Band event for SIGPOLL
```

Comment:
Since si_band is `OB_XSR` (Obsolete, XSI Streams) in the spec, and we are
using copy-in/copy-out semantics, I suggest this field be changed to a placeholder comment giving the name and that it is obsolete. That would
reduce complexity and cost spent on an obsolete field.  If done, a 
corresponding change to the underlying C code and perhaps any Ops
implicit.

Again, if you agree with the concept but do not have time/patience for the
change, let me know.

Reply:
I guess we could do this but we haven't in the past - we don't have any package docs to hold stuff like that.

Reply:
Not much in the way of docs - is the plan or do we want people to go the spec and stuff online? https://www.javadoc.io/doc/org.scala-native/posixlib_native0.4_2.11/latest/index.html#package

Reply:
I never knew or had see `www.javadoc.io/org.scala-native/posixlib_native`.  Learn something
every day.

I am usually a "document it all & do it early" person but my thinking is that at most this
should be documented as `experimental`.  I think it needs/deserves some more maturity
before even that.